modules/codegen/src/smithy4s/codegen/Renderer.scala

```diff
@@ -575,9 +575,10 @@ private[codegen] class Renderer(compilationUnit: CompilationUnit) { self =>
       hints: List[Hint]
   ): Lines = lines(
     block(
-      line"sealed abstract class $name(_value: String, _ordinal: Int) extends $Enumeration_.Value"
+      line"sealed abstract class $name(_value: String, _name: String, _ordinal: Int) extends $Enumeration_.Value"
```

Comment:
Couldn't decide on whether this should be optional or not.

Against optionality:

- smithy 2 enums seem to enforce names
- cleaner API, consistent usage among implementations that read this field

For optionality:

- smithy 1 enums don't enforce names (unless you provide at least one of them)

Reply:
Nah I'm cool with planning ahead.